captcha-validation-service/src/main/java/io/mosip/captcha/config/CommonConfig.java

```diff
@@ -8,9 +11,20 @@
 @Configuration
 public class CommonConfig {
 
+	@Value("${mosip.captcha.captcha-provider-name:#{null}}")
+	private String captchaProviderName;
+
 	@Primary
 	@Bean
 	public RestTemplate restTemplate() {
 		return new RestTemplate();
 	}
+
+	@Bean
+	public CaptchaProvider captchaProvider() {
```

Comment:
Create a CaptchaProviderFactory class. In this class autowire List<CaptchaProvider> providers. And in the captchaProvider interface create a method to return the provider name. Now based on the configured name you can iterate through the list of providers and return the provider matching the name. With this approach
1. each module can use its own captcha provider
2. Anyone can simply drop the implementation for captchaProvider in the classpath to use the new provider.